tests/python_package_test/test_dask.py

```diff
@@ -501,3 +821,29 @@ def f(part):
             model_factory=lgb.LGBMClassifier
         )
         assert 'foo' in str(info.value)
+
+
+@pytest.mark.parametrize(
+    "classes",
+    [
+        (lgb.DaskLGBMClassifier, lgb.LGBMClassifier),
+        (lgb.DaskLGBMRegressor, lgb.LGBMRegressor),
+        (lgb.DaskLGBMRanker, lgb.LGBMRanker)
+    ]
+)
+def test_dask_classes_and_sklearn_equivalents_have_identical_constructors_except_client_arg(classes):
+    def _compare_spec(dask_cls, sklearn_cls):
+        dask_spec = inspect.getfullargspec(dask_cls)
+        sklearn_spec = inspect.getfullargspec(sklearn_cls)
```

Comment:
I think it is possible to simplify here by removing inner function and setting directly 
```
dask_spec = inspect.getfullargspec(classes[0])
sklearn_spec = inspect.getfullargspec(classes[1])
```

Reply:
oh right! didn't think about how using `parametrize` meant this function was useless

Reply:
removed this function in https://github.com/microsoft/LightGBM/pull/3883/commits/876bfe5f6c756e946594e3316fec716ae358de18